src/oc_erchef/apps/oc_chef_authz/src/oc_chef_authz_acl.erl

```diff
@@ -31,6 +45,47 @@
                   reqid :: binary(),
                   darklaunch = undefined}).
 
+acl_spec(Part) ->
+    {[
+      {Part,
+       {[
+         {<<"groups">>, {array_map, string}},
+         {<<"actors">>, {array_map, string}},
+         {{opt, <<"clients">>}, {array_map, string}},
+         {{opt, <<"users">>}, {array_map, string}}
+        ]}}
+     ]}.
+
+% If client is present, user must be present
+% If user is present, client must be present.
+% If clients and users are present, actors must be an empty list.
+%
+% This is to ensure 'least surprise' - if you submit a
+% request containing actors, clients, and users it's not
+% going to be clear which one(s) are picked up for the update.
+%
+% This function assumes `acl_spec` has been applied in order to ensure
+% that `groups` and `actors`  are always present and that
+% the data types provided for each attribute are correct.
+validate_actors_clients_users(Part, FullACL) ->
```

Comment:
It might be nice to have a typespec here; I like to have specs on exported functions, but don't want to make you fix up all the tech debt.